src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -1433,3 +1480,64 @@ def merge_metadata(a_metadata, b_metadata, *, source, update=(), min_=(), max_=(
         result_directory=r_a.result_directory,
         app_version_info=app_version_info,
     )
+
+
+@contextmanager
+def open_global_sensitivity_analysis_result_file(results_directory: Path) -> h5py.File:
+    """
+    Open a global sensitivity analysis result file.
+    """
+    filename = results_directory / "uq_result"
+    ignored_file = results_directory / "uq_result.creating"
+    # Avoid to read result files with incomplete metadata.
+    if not ignored_file.is_file():
+        file = _open_result_file(filename)
+        yield file
+        file.close()
+    else:
+        yield
+
+
+def read_global_sensitivity_analysis_meta_data(
+    result_directory: Path,
+) -> Optional[UQResultMetadata]:
+    """
+    Read the global sensitivity analysis metadata persisted in a result file.
+    """
+
+    return UQResultMetadata.get_meta_data_from_file(result_directory=result_directory)
+
+
+def read_read_global_sensitivity_analysis_time_set(
+    result_directory: Path,
+) -> numpy.array:
+    """
+    Get the time set for sensitivity analysis results.
+    """
+    with open_global_sensitivity_analysis_result_file(
+        results_directory=result_directory
+    ) as result_file:
+        return result_file[GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME]["time_set"][:]
+
+
+def read_global_sensitivity_coefficients(
+    property_id: str, parametric_var_id: str, trend_id: str, results_directory: Path
+) -> Dict[str, numpy.array]:
+    """
+    Read the global sensitivity analysis coefficients results.
+    """
+    uq_meta_data = read_global_sensitivity_analysis_meta_data(results_directory)
+    with open_global_sensitivity_analysis_result_file(results_directory) as result_file:
+        global_sensitivity_analysis_coefficients = result_file[
+            GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME
+        ]["global_sensitivity_analysis"][:]
+
+    coefficients_key = f"{property_id}::{parametric_var_id}@{trend_id}"
+
+    coefficient_meta_data = uq_meta_data.global_gsa_meta_data[coefficients_key]
```

Comment:
If the file exists, they always have this key. As you can se in the ```SimFacade.SubmitUQResults```.
```python
SubmitUQResults(self, uq_results: UQSimulationResults) -> None:
        """
        Submit the uq results to a results file.
        :param uq_results:
            The uq results data.
        """
        import json

        import numpy

        from alfasim_sdk.result_reader.aggregator_constants import (
            GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME,
            META_GROUP_NAME,
            RESULTS_FOLDER_NAME,
            TIME_SET_DSET_NAME,
        )

        require_metadata_update = uq_results.require_metadata_update
        if require_metadata_update or (self._current_file is None):
            self.CloseResultFile()
            results_folder = self._directory / RESULTS_FOLDER_NAME
            results_folder.mkdir(parents=True, exist_ok=True)
            filename = 'uq_result'
            ignore_file = results_folder / (filename + '.creating')
            ignore_file.touch()
            self._CreateResultsFile(results_folder=results_folder, filename=filename)

            require_metadata_update = True

        assert self._current_file is not None
        f = self._current_file
        if require_metadata_update:
            meta_group = self._CreateMetaGroup(f)
            global_sensitivity_analysis_group = f.create_group(
                GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME, track_order=True
            )
            uq_meta_result_as_dict = {}
            for qoi_index, qoi_item in enumerate(uq_results.gsa_metadata):
                for result_index, result in enumerate(qoi_item):
                    new_key = (
                        f"{result['property']}::{result['parametric_var_id']}@{result['trend_id']}"
                    )
                    result['qoi_index'] = qoi_index
                    result['qoi_data_index'] = result_index
                    uq_meta_result_as_dict[new_key] = result

            meta_group.attrs['global_sensitivity_analysis'] = json.dumps(uq_meta_result_as_dict)
            global_sensitivity_analysis_group.create_dataset(
                TIME_SET_DSET_NAME, shape=(0,), dtype=numpy.float64, maxshape=(None,)
            )
            global_sensitivity_analysis_group.create_dataset(
                GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME,
                shape=(0, 0, 0),
                dtype=numpy.float64,
                maxshape=(None, None, None),
            )

            uq_results.require_metadata_update = False
            f.flush()
            f.swmr_mode = True
            ignore_file.unlink()

        else:
            meta_group = f[META_GROUP_NAME]
            global_sensitivity_analysis_group = f[GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME]

        if len(uq_results.global_sensitivity_coefficients) > 0:
            # GlobalSensitivityAnalysis / timeset ---------------------------------------------------------------------
            time_set_dset = global_sensitivity_analysis_group[TIME_SET_DSET_NAME]
            new_time_set_shape = uq_results.time_set.shape
            time_set_dset.resize(new_time_set_shape)
            time_set_dset[:] = uq_results.time_set
            # GlobalSensitivityAnalysis / GlobalSensitivityAnalysis ---------------------------------------------------
            gsa_data_set = global_sensitivity_analysis_group[GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME]
            gsa_data_set.resize(uq_results.global_sensitivity_coefficients.shape)
            gsa_data_set[:] = uq_results.global_sensitivity_coefficients

        f.flush()
```

Reply:
I see, but note that the variables that compose the `coefficient_key` string are all received as arguments, so a client of this function could easily pass something that doesn't match any key, e.g. with typos on `property_id` or simply because the caller expects some key on the results that for any reason doesn't exist.

But again, this would be a more cautious approach, to handle border cases that in fact could never blow up, so it's up to you. Personally I like to take a more "defensive" approach, to cope with weird scenarios (or at least making then more visible) beforehand.

Reply:
@ro-oliveira95 has a point, "this can never occur today because of this code" is not good enough, because "this code" might change in the future, where you assumptions are no longer valid.

Having said that, I think it is fine to just let this blow up with whatever exception will happen when we try to access an invalid group.

Reply:
Technically the key should not be constructed. It should come from the existing metadata. If it is in the metadata the group/dset should exist.

This design is not an implementation detail. Is the protocol.

Those id's have no clear meaning to sdk. It is actually possible to access the metadata and get then, but we usually offer a high level interface to abstract that:

https://github.com/ESSS/alfasim-sdk/blob/master/src/alfasim_sdk/result_reader/reader.py#L147-L170

That `trend_metadata` has those ids. But most ids (property is the exception) are internal alfasim detailts.